### ISSUE TITLE
Better error reporting for `--frozen-lockfile`

### ISSUE THREAD

**Issue by silversquirl** [oven-sh/bun#5921]
### What is the problem this feature would solve?

The current error (`error: lockfile had changes, but lockfile is frozen`) is not very helpful. It indicates that there *is* a problem, but not what the problem is or how to solve it.

### What is the feature you are proposing to solve the problem?

It would be much nicer if, in case of a conflict, bun were to output the names of the packages causing the conflict, as well as the expected version from the lockfile and the version range specified in `package.json`.

### What alternatives have you considered?

The only alternative seems to be manually comparing versions between `bun.lockb` and `package.json`, since even `--verbose` outputs no useful information here, however that is extremely tedious and often impractical.

**Comment by baptisteArno** [comment]
Yes, I'm just clueless here. In my case it happens in a docker build process. I copy all the files and run the install command but it fails with that error.

```
COPY . /temp/dev/
RUN cd /temp/dev && bun install --frozen-lockfile
```

The same command works on my machine (outside docker context)
